dockerfiles/compilation/java-8/integration_tests.sh

```diff
@@ -52,6 +52,4 @@ export JAMES_IMAP_PORT=$JAMES_IMAP_PORT
 git clone $ORIGIN/.
 git checkout $SHA1
 
-
-mvn -DskipTests -pl org.apache.james:apache-james-mpt-external-james -am compile
-mvn -Dtest=JamesDeploymentValidation -pl org.apache.james:apache-james-mpt-external-james test
+mvn compile test -Dtest=JamesDeploymentValidation -DfailIfNoTests=false -am -pl org.apache.james:apache-james-mpt-external-james
```

Comment:
can you explain why we do need this?

Reply:
Without the `failIfNoTests` option, the build will fail on the first _dependency_ project because the `JamesDeploymentValidation` test will not be found (it's in the last project built).

Reply:
my question is related to matthieu's remark: why do you need to cache artifact / or build everything at once?

Reply:
There are two kinds of dependencies:
- external: download them from maven central & store them in maven volume
- internal: (your maven graph) should be built each time by mvn commands & we don't need to store them in maven volume

This part of code was wrong from the beginning, as the first `mvn` command is not deploying artifacts built, the second `mvn` command is always downloading them from maven central.

In my current test, it's failing because the version I'm trying to build has never been uploaded in maven central.

There are multiple solutions:
- deploy the artifacts in maven volume (only 15 projects)
 PROS: easiest solution
 CONS: things related to one build are remaining in the maven volume
- use one `mvn` command (with failIfNoTests)
 PROS: one command
 CONS: if the test is not played it will not fail...
- try other `mvn` goals
 WIP

Reply:
a better solution would be to actually install built artifacts but in a directory related to this container (so not in the VOLUME).
We can very well have a multi-stage build for that:
1. first Dockerfile step only retrieve dependencies
2. then we make next steps from this image